sst/SimengCoreWrapper.cc

```diff
@@ -0,0 +1,253 @@
+                      endTime - startTime_)
+                      .count();
+  double hz = iterations_ / (static_cast<double>(duration) / 1000.0);
+  double khz = hz / 1000.0;
+  uint64_t retired = core_->getInstructionsRetiredCount();
+  double mips = retired / static_cast<double>(duration) / 1000.0;
+
+  // Print stats
+  std::cout << "\n";
+  auto stats = core_->getStats();
+  for (const auto& [key, value] : stats) {
+    std::cout << key << ": " << value << "\n";
+  }
+
+  std::cout << "\nFinished " << iterations_ << " ticks in " << duration
+            << "ms (" << std::round(khz) << " kHz, " << std::setprecision(2)
+            << mips << " MIPS)" << std::endl;
+
+  delete[] processMemory_;
+}
+
+void SimengCoreWrapper::init(unsigned int phase) {
+  mem_->init(phase);
+  // Init can have multiple phases, only fabricate the core once at phase 0
+  if (phase == 0) {
+    fabricateSimengCore();
+  }
+}
+
+bool SimengCoreWrapper::clockTick(SST::Cycle_t current_cycle) {
+  // Tick the core and memory interfaces until the program has halted
+  if (!core_->hasHalted() || dataMemory_->hasPendingRequests()) {
+    // Tick the core
+    core_->tick();
+    // Update Virtual Counter Timer at correct frequency.
+    if (iterations_ % (uint64_t)timerModulo_ == 0) {
+      vitrualCounter_++;
+      core_->incVCT(vitrualCounter_);
+    }
+
+    // Tick memory
+    instructionMemory_->tick();
+    dataMemory_->tick();
+
+    iterations_++;
+
+    return false;
+  } else {
+    // Protected method from SST::Component used to end SST simulation
+    primaryComponentOKToEndSim();
+    return true;
+  }
+}
+
+void SimengCoreWrapper::fabricateSimengCore() {
+  output_.verbose(CALL_INFO, 1, 0, "Setting up SimEng Core\n");
+
+  SimulationMode mode = SimulationMode::InOrderPipelined;
+  std::string modeString;
+  YAML::Node config;
+
+  if (configPath_ != "") {
+    config = simeng::ModelConfig(configPath_).getConfigFile();
+  } else {
+    config = YAML::Load(DEFAULT_CONFIG);
+  }
+
+  if (config["Core"]["Simulation-Mode"].as<std::string>() == "inorder") {
+    output_.fatal(CALL_INFO, 1, 0,
+                  "SimEng SST build does not support in-order mode yet!\n");
+  }
+
+  if (config["Core"]["Simulation-Mode"].as<std::string>() == "outoforder") {
+    mode = SimulationMode::OutOfOrder;
+    modeString = "Out-of-Order";
+  } else {
+    mode = SimulationMode::Emulation;
+    modeString = "Emulation";
+  }
+
+  float clockFreq_ = config["Core"]["Clock-Frequency"].as<float>();
+  uint32_t timerFreq_ = config["Core"]["Timer-Frequency"].as<uint32_t>();
+  timerModulo_ = (clockFreq_ * 1e9) / (timerFreq_ * 1e6);
+
+  // Create the process Image
+  std::vector<std::string> commandLine({executablePath_, executableArgs_});
+  process_ =
+      std::make_unique<simeng::kernel::LinuxProcess>(commandLine, config);
+  if (!process_->isValid())
+    output_.fatal(CALL_INFO, 1, 0, "Could not read/parse %s", executablePath_);
```

Comment:
To fix the AppleClang error `error: cannot pass object of non-trivial type 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') through variadic method; call will abort at runtime [-Wnon-pod-varargs]`, change `executablePath_` to `executablePath_.c_str()`